<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigDialog</class>
 <widget class="QDialog" name="ConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>970</width>
    <height>643</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="modal">
   <bool>false</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="applicationTab">
      <attribute name="title">
       <string>GitKlient</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="frame">
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label_21">
              <property name="text">
               <string>Editor font family</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QFontComboBox" name="cbEditorFontFamily"/>
              </item>
              <item>
               <widget class="QLabel" name="label_23">
                <property name="text">
                 <string>size</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="sbEditorFontSize">
                <property name="suffix">
                 <string> pt</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Commit title max. length</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spCommitTitleLength">
              <property name="value">
               <number>50</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Styles schema</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="cbStyle">
              <item>
               <property name="text">
                <string>dark</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>bright</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Git Location (if not in PATH)</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="leGitPath">
              <property name="placeholderText">
               <string>Git location...</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Logs</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QCheckBox" name="chEnableLogs">
                <property name="text">
                 <string>Enabled</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>log level</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="cbLogLevel">
                <item>
                 <property name="text">
                  <string>Trace</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Debug</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Info</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Warning</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Error</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Fatal</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="repositoryTab">
      <attribute name="title">
       <string>Repository</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>925</width>
            <height>794</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QFrame" name="frame_2">
             <layout class="QGridLayout" name="gridLayout">
              <property name="leftMargin">
               <number>10</number>
              </property>
              <property name="topMargin">
               <number>10</number>
              </property>
              <property name="rightMargin">
               <number>10</number>
              </property>
              <property name="bottomMargin">
               <number>10</number>
              </property>
              <property name="spacing">
               <number>10</number>
              </property>
              <item row="5" column="1">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>The interval is expected to be in minutes. Choose a value between 0 (for disabled) and 60.</string>
                </property>
                <property name="wordWrap">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="9" column="0">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Cache space used:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Order of commits in the graph</string>
                </property>
               </widget>
              </item>
              <item row="8" column="1">
               <widget class="QCheckBox" name="clangFormat">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="8" column="0">
               <widget class="QLabel" name="labelClangFormat">
                <property name="text">
                 <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Run clang-format before commit&lt;br/&gt;(Requires clang-format)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_20">
                <property name="text">
                 <string>Pull strategy:</string>
                </property>
               </widget>
              </item>
              <item row="16" column="1">
               <widget class="QCheckBox" name="cbDeleteFolder">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="13" column="1">
               <widget class="QCheckBox" name="cbStash">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="14" column="0">
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>Show submodules</string>
                </property>
               </widget>
              </item>
              <item row="9" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout">
                <property name="spacing">
                 <number>10</number>
                </property>
                <item>
                 <widget class="QLabel" name="lCacheSize">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="ButtonLink" name="pbClearCache">
                  <property name="text">
                   <string>Clear</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item row="16" column="0">
               <widget class="QLabel" name="label_22">
                <property name="text">
                 <string>Delete remote branches folder</string>
                </property>
               </widget>
              </item>
              <item row="14" column="1">
               <widget class="QCheckBox" name="cbSubmodule">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QComboBox" name="cbLogOrder">
                <item>
                 <property name="text">
                  <string>Author date order</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Date order</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Topo order</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="sbMaxCommits">
                <property name="suffix">
                 <string> last commits</string>
                </property>
                <property name="maximum">
                 <number>999999999</number>
                </property>
                <property name="singleStep">
                 <number>10</number>
                </property>
                <property name="value">
                 <number>0</number>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QComboBox" name="cbPullStrategy">
                <item>
                 <property name="text">
                  <string>Merge (default)</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Rebase</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Fast-forward only</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="6" column="1">
               <widget class="QCheckBox" name="pruneOnFetch">
                <property name="text">
                 <string/>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="11" column="0">
               <widget class="QLabel" name="label_24">
                <property name="text">
                 <string>Show remote branches</string>
                </property>
               </widget>
              </item>
              <item row="18" column="0">
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>Auto-Fetch interval</string>
                </property>
               </widget>
              </item>
              <item row="7" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Update submodules when pull</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QSpinBox" name="autoFetch">
                <property name="suffix">
                 <string> minutes</string>
                </property>
                <property name="maximum">
                 <number>60</number>
                </property>
                <property name="value">
                 <number>5</number>
                </property>
               </widget>
              </item>
              <item row="10" column="0">
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>Show local branches</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="cbTranslations"/>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Language</string>
                </property>
               </widget>
              </item>
              <item row="15" column="1">
               <widget class="QCheckBox" name="cbSubtree">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="13" column="0">
               <widget class="QLabel" name="label_16">
                <property name="text">
                 <string>Show stashes</string>
                </property>
               </widget>
              </item>
              <item row="7" column="1">
               <widget class="QCheckBox" name="updateOnPull">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Max. commits to retrieve (0 for all)</string>
                </property>
               </widget>
              </item>
              <item row="6" column="0">
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Prune when fetch</string>
                </property>
               </widget>
              </item>
              <item row="17" column="0" colspan="2">
               <widget class="QGroupBox" name="credentialsFrames">
                <property name="title">
                 <string>Credentials configuration</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_4">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <property name="spacing">
                  <number>10</number>
                 </property>
                 <item row="0" column="0" colspan="2">
                  <widget class="QCheckBox" name="chbCredentials">
                   <property name="text">
                    <string>Store credentials using GitQlient (Current session or in Storage)</string>
                   </property>
                   <property name="checked">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QRadioButton" name="rbStorage">
                   <property name="text">
                    <string>Storage</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QRadioButton" name="rbCache">
                   <property name="text">
                    <string>Cache for current session</string>
                   </property>
                   <property name="checked">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <layout class="QHBoxLayout" name="horizontalLayout_3">
                   <property name="spacing">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="label_19">
                     <property name="text">
                      <string>TextLabel</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QSpinBox" name="sbTimeout">
                     <property name="suffix">
                      <string> mins.</string>
                     </property>
                     <property name="maximum">
                      <number>480</number>
                     </property>
                     <property name="singleStep">
                      <number>5</number>
                     </property>
                     <property name="value">
                      <number>15</number>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item row="2" column="1">
                  <widget class="QPushButton" name="pbAddCredentials">
                   <property name="text">
                    <string>Add credentials</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="15" column="0">
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>Show subtrees</string>
                </property>
               </widget>
              </item>
              <item row="12" column="0">
               <widget class="QLabel" name="label_25">
                <property name="text">
                 <string>Show tags</string>
                </property>
               </widget>
              </item>
              <item row="10" column="1">
               <widget class="QCheckBox" name="cbLocal">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="11" column="1">
               <widget class="QCheckBox" name="cbRemote">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="12" column="1">
               <widget class="QCheckBox" name="cbTags">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="gitConfigTab">
      <attribute name="title">
       <string>Git Config</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QTabWidget" name="gitConfigTabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="localGit">
          <attribute name="title">
           <string>Local</string>
          </attribute>
         </widget>
         <widget class="QWidget" name="globalGit">
          <attribute name="title">
           <string>Global</string>
          </attribute>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ButtonLink</class>
   <extends>QLabel</extends>
   <header>ButtonLink.hpp</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>chEnableLogs</tabstop>
  <tabstop>spCommitTitleLength</tabstop>
  <tabstop>cbLogLevel</tabstop>
  <tabstop>cbStyle</tabstop>
  <tabstop>leGitPath</tabstop>
  <tabstop>scrollArea</tabstop>
  <tabstop>cbTranslations</tabstop>
  <tabstop>sbMaxCommits</tabstop>
  <tabstop>cbLogOrder</tabstop>
  <tabstop>autoFetch</tabstop>
  <tabstop>pruneOnFetch</tabstop>
  <tabstop>updateOnPull</tabstop>
  <tabstop>clangFormat</tabstop>
  <tabstop>cbStash</tabstop>
  <tabstop>cbSubmodule</tabstop>
  <tabstop>cbSubtree</tabstop>
  <tabstop>chbCredentials</tabstop>
  <tabstop>rbCache</tabstop>
  <tabstop>rbStorage</tabstop>
  <tabstop>sbTimeout</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <slots>
  <slot>saveConfig()</slot>
 </slots>
</ui>
